### PR TITLE
Show a tooltip explaining why the run button is initially disabled for contained levels

### PR DESCRIPTION
**Pull request by balderdash**
It shows up if you try to click on the disabled run button.

image

### CODE REVIEW

**Comment by joshlory** [comment]
What do you think about adding an extra callout to `appOptions.callouts` with an on event for the run button click?  See: https://github.com/code-dot-org/code-dot-org/pull/664.

Might end up being messy to dynamically add the callout for contained levels...

**Comment by balderdash** [comment]
Switched to a callout.

I just called `addCallouts` directly instead of modifying `appOptions`, since I'd rather not assume that the contained level initialization will always happen before the normal callouts are created.

File: apps/src/templates/ContainedLevel.jsx [hunk 1]
```diff
@@ -31,7 +33,37 @@ const ContainedLevel = React.createClass({
...
+      let boundingClientRect = runButton.get(0).getBoundingClientRect();
+      clickReceiver.css({
+        height: boundingClientRect.height + 'px',
+        width: boundingClientRect.width + 'px',
+        position: 'absolute',
+        top: 0,
+      });
+      $('#gameButtons').append(clickReceiver);
+      clickReceiver.bind('click',
+        () => $(window).trigger('attemptedRunButtonClick'));
```
Comment (joshlory) [inline apps/src/templates/ContainedLevel.jsx]: Nit: I think you can re-use the [`run_button_pressed` event](https://github.com/code-dot-org/code-dot-org/blob/ef0a7ad16856f7f6f5c6b22200be0552c88e7c5b/apps/src/StudioApp.js#L1691).

File: apps/src/templates/ContainedLevel.jsx [hunk 2]
```diff
@@ -31,7 +33,37 @@ const ContainedLevel = React.createClass({
...
+
+      addCallouts([{
```
Comment (joshlory) [inline apps/src/templates/ContainedLevel.jsx]: I wonder if this makes sense to do elsewhere in contained level init?  It's a bit weird to have this as a side effect of mounting the component.

If I unmount and re-mount the component multiple times (e.g. in Storybook) do I get multiple callouts?

File: apps/src/templates/ContainedLevel.jsx (hunk 1 above)
Comment (joshlory) [reply apps/src/templates/ContainedLevel.jsx]: Ah never mind, I see that the Run Button is disabled.  I wonder if [capturing events on the parent element](https://stackoverflow.com/a/32059136) would be cleaner?  Or not disabling the button, but bailing out early in the click handler if there's no answer selected.

File: apps/src/code-studio/callouts.js [hunk 3]
```diff
@@ -71,9 +67,36 @@ module.exports = function createCallouts(callouts) {
     showHideDropletGutterCallouts();
   });
 
-  var showCalloutsMode = document.URL.indexOf('show_callouts=1') !== -1;
 
   $.fn.qtip.zindex = 500;
+  addCallouts(callouts);
+
+  // Insert a hashchange handler to detect triggercallout= hashes and fire
+  // appropriate events to open the callout
```
Comment (joshlory) [inline apps/src/code-studio/callouts.js:75]: I was curious why we have this code.  An interesting bit of history!  https://github.com/code-dot-org/code-dot-org/pull/8838/files#r66492285

File: apps/src/templates/ContainedLevel.jsx (hunk 1 above)
Comment (balderdash) [reply apps/src/templates/ContainedLevel.jsx]: That's much nicer, removed the dummy element.

File: apps/src/templates/ContainedLevel.jsx (hunk 2 above)
Comment (balderdash) [reply apps/src/templates/ContainedLevel.jsx]: I moved the contained level init call to `StudioApp.init()`.

**Pull request merged**